### ISSUE TITLE
Properly Handle Modal Visibility

### ISSUE THREAD

**Issue by joshmorel** [chingu-coders/voyage1-turtles-team-38#21]
## Overview

We have many modals (pop-ups) for which visibility can be controlled in several ways.

## Cases 

In Leoh, all are hidden upon opening by default. The visibility can be changed via:

* Case 1: If associated icon for that modal is clicked the modal should be opened
* Case 2: If modal is open, when anything in the window EXCEPT that same modal is clicked then the modal should be hidden
* Case 3: If option is set to for modal to be active on open, it should be displayed on page load (requires localStorage)

Case 1 & 3 are simple and existing architecture can handle them.

Case 2 is unfortunately quite complicated. As an example, see this popup implementation in [Semantic UI](https://github.com/Semantic-Org/Semantic-UI/blob/577ae996568e355cb0141a7cce9ba07bec2ceceb/src/definitions/modules/popup.js)

### Option 1 - track "active" modal: 

* In Redux, track attribute top-level activeModal variable
* Each modal is passed action which will change the modal from active to non-active based on mouseenter and mouseleave listeners
* When any part of window is clicked, this triggers an action to hide non-active modals
* Reducer changes all modals to hidden EXCEPT the active modal which is tracked via state

**Comment by joshmorel** [comment]
We may also want to consider at this point renaming of modal to "pop-up" or something else. Modal is more like actually the google search widget which can only be closed upon pressing X.

Options are:

* pop-up
* tool (this is what they are referred to in options modal/pop-up/tool)

**Comment by wjhurley** [comment]
Is Case 2 complicated because we're using React? Looking at Leoh, the CSS display property is just toggled from "block" to "none" when clicking outside of the modal/popup. Is there an easy way we can tie an event listener to the whole page except for the modal/popup?

**Comment by joshmorel** [comment]
I wouldn't say so. It's more because as far as I can tell there seems to be no simple way to do this without a library. If you look on about line 726 in `load.js` they've added whole page event listener for each modal/popup/tool. The jQuery function [closest](https://api.jquery.com/closest/) is used to test if the user is clicking within the icon or modal. If the user's clicking within the icon it opens the modal (using slideToggle), if the user is NOT clicking in modal (basically anywhere else) it closes the modal (using slideUp). Not sure if this is clever, ugly or both, but it does work. 

```
//    $("#news").hide();
    $(document).on('click', function(e) {
        if ($(e.target).closest('#newsIcon').length) {
            $("#news").slideToggle();
            ga('send', 'event', 'news', 'click', 'opened');
        } else if (!$(e.target).closest('#news').length) {
            $('#news').slideUp();
        }
    });
```

We could do something similar, but I think we're best to leverage a React component library, I found at least one option there could be others.
 
Some options and resources:
* [react-onclickoutside](https://github.com/Pomax/react-onclickoutside)
* [closest library](https://www.npmjs.com/package/closest)
* [adding document event listener with React](https://discuss.reactjs.org/t/how-to-calculate-jqyery-closest-in-react/1988/4)
* Approach 1 above may work... Not sure

**Comment by joshmorel** [comment]
I just tried this one and it's perfect: [react-click-outside](https://github.com/tj/react-click-outside)

I'll also try the more starred react-onclickoutside listed earlier (practically the same name, but different projects).

**Comment by joshmorel** [comment]
Resolved by #25
